### PR TITLE
modify the dataset format to relative path

### PR DESCRIPTION
**Pull request by jaypume**

### CODE REVIEW

File: lib/sedna/dataset/dataset.py [hunk 1]
```diff
@@ -18,71 +18,53 @@
...
+        if preprocess_fun:
+            new_samples = [preprocess_fun(s) for s in samples]
         else:
-            return _load_txt_dataset(dataset_url)
+            new_samples = samples
+        return new_samples
 
 
-def load_train_dataset(data_format, **kwargs):
+def load_train_dataset(data_format, preprocess_fun, **kwargs):
```
Comment (llhuii) [inline lib/sedna/dataset/dataset.py:44]: this adds a extra preprocess_fun arg, how about code example's compatibility? such as the incremental learning example.

Comment (jaypume) [reply lib/sedna/dataset/dataset.py]: In the incremental learning example,  it works fine with keeping preprocess_fun empty.

File: lib/sedna/dataset/dataset.py [hunk 2]
```diff
@@ -18,71 +18,53 @@
...
+        if preprocess_fun:
+            new_samples = [preprocess_fun(s) for s in samples]
         else:
-            return _load_txt_dataset(dataset_url)
+            new_samples = samples
+        return new_samples
 
 
-def load_train_dataset(data_format, **kwargs):
+def load_train_dataset(data_format, preprocess_fun=None, **kwargs):
```
Comment (llhuii) [inline lib/sedna/dataset/dataset.py:44]: why not just our load_tran_dataset just return a list of files, instead of add a preprocess_fun?

**Comment by llhuii** [comment]
can you squash the commits?

File: lib/sedna/dataset/dataset.py (hunk 2 above)
Comment (jaypume) [reply lib/sedna/dataset/dataset.py:44]: I think the sample list returned could be more flexible, as long as it conforms to the training interface written by user.

**Pull request merged**